eos-repartition-mbr

```diff
@@ -0,0 +1,171 @@
+#!/bin/bash -e
+# Copyright (C) 2016 Endless Mobile, Inc.
+# Licensed under the GPLv2
+
+# arguments checking
+if [ $# -lt 1 ]; then
+	echo "Usage: $0 <EOS partition>"
+	exit 1
+fi
+
+root_part=$1
+root_partno=${root_part: -1}
+root_disk=${root_part%?}
+
+swap_partno=$((root_partno + 1))
+swap_part=${root_disk}${swap_partno}
+
+# check if we passed the full device file path
+if [ "${root_part//dev/}" = "${root_part}" ]; then
+	echo "$0: Specify the full path for the root device file (i.e. /dev/sda3)"
+	exit 1
+fi
+
+# check if we passed the device file for the partition and not for the disk
+case ${root_partno} in
+	*[!0-9]* )
+		echo "$0: Specify the root partition (i.e. /dev/sda3)"
+		exit 1
+		;;
+esac
+
+# check if the root partition does exist (if it exists we assume $root_disk does exist too)
+if [ ! -e "${root_part}" ]; then
+  echo "$0: $root_part doesn't exist"
+  exit 1
+fi
+
+# udev might still be busy probing the disk, meaning that it will be in use.
+udevadm settle
+
+# take current partition table
+parts=$(sfdisk -d $root_disk)
+
+# take the root partition type GUID
+root_part_dev=${root_part#/dev/}
+root_guid=$(echo "$parts" | sed -n -e "s/.*$root_part_dev.*type=\([A-Z0-9-]*\).*/\1/p")
```

Comment:
yes, and finding `${root_disk}` can be done more cleanly with `lsblk -o PKNAME`, so I can kill at least two birds with one patch